cpp/src/strings/reverse.cu

```diff
@@ -62,10 +63,10 @@ std::unique_ptr<column> reverse(strings_column_view const& input,
   if (input.is_empty()) { return make_empty_column(type_id::STRING); }
 
   // copy the column; replace data in the chars column
-  auto result = std::make_unique<column>(input.parent(), stream, mr);
-  auto const d_offsets =
-    result->view().child(strings_column_view::offsets_column_index).data<size_type>();
-  auto d_chars = result->mutable_view().head<char>();
+  auto result          = std::make_unique<column>(input.parent(), stream, mr);
+  auto sv              = strings_column_view(result->view());
```

Comment:
Could `result` be `const` here?

Reply:
we do modify the column, const would be a bit misleading IMO :shrug:

Reply:
FWIW, constness does not interfere with the copy elision, e.g. https://godbolt.org/z/TzPjshqrd
We'd need a `std::move` in the return if the copy elision didn't kick in, since `result` is not copyable.

I agree that const sends the wrong message regardless.